### PR TITLE
CRUD prometheus rule groups API for cortexadmin Plugin

### PR DESCRIPTION
**Pull request by alexandreLamarre**
- [x] CRUD prom rules for cortexadmin
- [x] Testdata + test suite for cortexadmin plugin prom rules CRUD

### CODE REVIEW

File: plugins/cortex/pkg/apis/cortexadmin/cortexadmin.proto [hunk 1]
```diff
@@ -37,6 +37,34 @@ service CortexAdmin {
       }
     };
   }
+  rpc GetRule(RuleRequest) returns (QueryResponse) {
+    option (google.api.http) = {
+      get: "/get_rule/{groupName}"
```
Comment (kralicky) [inline plugins/cortex/pkg/apis/cortexadmin/cortexadmin.proto:42]: To follow proper REST style, these should be:
```
GET /rules/{groupName}
```

File: plugins/cortex/pkg/apis/cortexadmin/cortexadmin.proto [hunk 2]
```diff
@@ -37,6 +37,34 @@ service CortexAdmin {
...
+      additional_bindings {
+        post: "/get_rule/{groupName}"
+        body: "*"
+      }
+    };
+  }
+
+  rpc LoadRules(YamlRequest) returns (google.protobuf.Empty) {
+    option (google.api.http) = {
+      get: "/load_rules"
```
Comment (kralicky) [inline plugins/cortex/pkg/apis/cortexadmin/cortexadmin.proto:52]: (cont)
```
POST /rules/{groupName}
```

File: plugins/cortex/pkg/apis/cortexadmin/cortexadmin.proto [hunk 3]
```diff
@@ -37,6 +37,34 @@ service CortexAdmin {
...
+      additional_bindings {
+        post: "/load_rules"
+        body : "*",
+      }
+    };
+  }
+  rpc DeleteRule(RuleRequest) returns (google.protobuf.Empty) {
+    option (google.api.http) = {
+      get: "/delete_rules/{groupName}"
```
Comment (kralicky) [inline plugins/cortex/pkg/apis/cortexadmin/cortexadmin.proto:61]: (cont)
```
DELETE /rules/{groupName}
```

**Review by kralicky — CHANGES_REQUESTED** [review]

File: plugins/cortex/pkg/apis/cortexadmin/cortexadmin.proto (hunk 1 above)
Comment (alexandreLamarre) [reply plugins/cortex/pkg/apis/cortexadmin/cortexadmin.proto]: Done!

File: plugins/cortex/pkg/apis/cortexadmin/cortexadmin.proto (hunk 2 above)
Comment (alexandreLamarre) [reply plugins/cortex/pkg/apis/cortexadmin/cortexadmin.proto]: Done!

File: plugins/cortex/pkg/apis/cortexadmin/cortexadmin.proto (hunk 3 above)
Comment (alexandreLamarre) [reply plugins/cortex/pkg/apis/cortexadmin/cortexadmin.proto]: Done!

**Review by kralicky — APPROVED** [review]

**Pull request merged**